Solver/elastic_law.py

```diff
@@ -0,0 +1,310 @@
+import shenfun as sf
+import sympy as sp
+import numpy as np
+import itertools as it
+from shenfun import inner, div, grad, Dx
+
+
+class LinearCauchyElasticity:
+    def __init__(self):
+        self.name = "LinearCauchyElasticity"
+        self.nb_material_parameters = 2
+
+    def check_with_pde(self, u_hat, material_parameters, body_forces):
+        assert isinstance(u_hat, sf.Function)
+        assert len(material_parameters) == self.nb_material_parameters
```

Comment:
Shouldn't there be an assertion to guarantee that `body_forces` has a compatible type?